stdlib.sh

```diff
@@ -1202,9 +1211,14 @@ on_git_branch() {
     return 1
   elif [ -z "$1" ]; then
     return 0
+  elif [ "$1" = "-r" ] && [ -z "$2" ]; then
+    log_error "missing regexp pattern after \`-r\` flag"
+    return 1
   fi
   watch_file "$git_dir/HEAD"
-  [ "$(git branch --show-current)" = "$1" ]
+  local git_branch
+  git_branch=$(git branch --show-current)
+  [ "$1" = "$git_branch" ] || [ "$1" = '-r' ] && [[ "$git_branch" =~ $2 ]]
```

Comment:
The logic would work unless `"$1" = "-r"`. This is unlikely but it's better to use `if [[ "$1 = '-r' ]]; then [[ $git_branch" =~ $2 ]] else [[ "$1" = "$git_branch" ]]; fi`